tests/inc/az_ulib_ustream_forward_compliance_ut.h

```diff
@@ -0,0 +1,612 @@
+
+  /// cleanup
+  (void)result;
+}
+
+/* The get_size shall return the number of bytes between the current position and the end
+ * of the buffer. */
+static void az_ulib_ustream_forward_get_size_compliance_new_buffer_succeed(void** state)
+{
+  /// arrange
+  (void)state;
+  az_ulib_ustream_forward* ustream_forward;
+  USTREAM_FORWARD_COMPLIANCE_TARGET_FACTORY(&ustream_forward);
+
+  /// act
+  size_t size = az_ulib_ustream_forward_get_size(ustream_forward);
+
+  /// assert
+  assert_int_equal(size, USTREAM_FORWARD_COMPLIANCE_EXPECTED_CONTENT_LENGTH);
+
+  /// cleanup
+  az_result result = az_ulib_ustream_forward_dispose(ustream_forward);
+  (void)result;
+}
+
+/* [1]The read shall copy the content in the provided buffer and return the number of valid
+ * <tt>uint8_t</tt> values in the local buffer in the provided `size`. */
+/* [2]If the length of the content is bigger than the `buffer_length`, the read shall limit the copy
+ * size to the buffer_length.*/
+/* [3]If there is no more content to return, the read shall return AZ_ERROR_ITEM_NOT_FOUND,
+ * size shall receive 0, and do not change the content of the local buffer. */
+static void az_ulib_ustream_forward_read_compliance_get_from_original_buffer_succeed(void** state)
+{
+  /// arrange
+  (void)state;
+  az_ulib_ustream_forward* ustream_forward;
+  USTREAM_FORWARD_COMPLIANCE_TARGET_FACTORY(&ustream_forward);
+  uint8_t buf_result1[USTREAM_FORWARD_COMPLIANCE_TEMP_BUFFER_LENGTH];
+  uint8_t buf_result2[USTREAM_FORWARD_COMPLIANCE_TEMP_BUFFER_LENGTH];
+  uint8_t buf_result3[USTREAM_FORWARD_COMPLIANCE_TEMP_BUFFER_LENGTH];
+  size_t size_result1;
+  size_t size_result2;
+  size_t size_result3;
+
+  /// act
+  az_result result1 = az_ulib_ustream_forward_read(
+      ustream_forward, buf_result1, USTREAM_FORWARD_COMPLIANCE_LENGTH_1, &size_result1);
+  az_result result2 = az_ulib_ustream_forward_read(
+      ustream_forward, buf_result2, USTREAM_FORWARD_COMPLIANCE_TEMP_BUFFER_LENGTH, &size_result2);
+  az_result result3 = az_ulib_ustream_forward_read(
+      ustream_forward, buf_result3, USTREAM_FORWARD_COMPLIANCE_TEMP_BUFFER_LENGTH, &size_result3);
+
+  /// assert
+  assert_int_equal(result1, AZ_OK);
+  assert_int_equal(size_result1, USTREAM_FORWARD_COMPLIANCE_LENGTH_1);
+  assert_memory_equal(USTREAM_FORWARD_COMPLIANCE_LOCAL_EXPECTED_CONTENT, buf_result1, size_result1);
+
+  assert_int_equal(result2, AZ_OK);
+  assert_int_equal(
+      size_result2,
+      USTREAM_FORWARD_COMPLIANCE_EXPECTED_CONTENT_LENGTH - USTREAM_FORWARD_COMPLIANCE_LENGTH_1);
+  assert_memory_equal(
+      (const uint8_t* const)(
+          USTREAM_FORWARD_COMPLIANCE_LOCAL_EXPECTED_CONTENT + USTREAM_FORWARD_COMPLIANCE_LENGTH_1),
+      buf_result2,
+      size_result2);
+
+  assert_int_equal(result3, AZ_ULIB_EOF);
+
+  /// cleanup
+  az_result result = az_ulib_ustream_forward_dispose(ustream_forward);
+  (void)result;
+}
+
+static void az_ulib_ustream_forward_flush_compliance_single_buffer_succeed(void** state)
```

Comment:
>az_ulib_ustream_forward_flush_compliance_single_buffer_succeed [](http://example.com/codeflow?start=12&length=62)

Please add a test for `flush` after `read`.

Reply:
I mean, for example, in a ustream_forward with 20 bytes, read the first 5 bytes with `read` and flush the rest with `flush`.

Reply:
can do. implemented an e2e test for this